### PR TITLE
Fix NPE when listing schemas in Phoenix connector

### PR DESCRIPTION
**Pull request by ebyhr**

### CODE REVIEW

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java [hunk 1]
```diff
@@ -193,8 +195,11 @@ public BaseJdbcClient(
         }
     }
 
-    protected boolean filterSchema(String schemaName)
+    protected boolean filterSchema(@Nullable String schemaName)
```
Comment (losipiuk) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java:198]: Is there an issue it fixes? What is the case when null is passed here?

Comment (findepi) [reply presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java:198]: @losipiuk Phoenix, see @lhofhansl 's https://prestosql.slack.com/archives/CP1MUNEUX/p1595558846274400 

@ebyhr 
When a connector reports some tables with null schema, there are inaccessible
in current base jdbc implementation. To properly handle them you'd need to change
more places besides this one. Thus, I would consider null schemas a concrete
connnector responsibility until we determine how frequent or how problematic this
approach is.

**Comment by findepi** [comment]
Is it possible to cover this with a test?

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java (hunk 1 above)
Comment (findepi) [reply presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java:198]: quoting @lhofhansl 
> In Phoenix tables do not need to have a schema. For that the connector adds a virtual "default" schema.

this means it's not OK to filter them out.
perhaps the Phoenix code could provide `firstNonNull(schemaname, "default")` here.

Also, this probably means we should be able to test those null schemas.

**Comment by ebyhr** [comment]
@findepi I'm looking into the way to test.

**Comment by lhofhansl** [comment]
@vincentpoon Do you have a good idea about how to add a test?

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java (hunk 1 above)
Comment (lhofhansl) [reply presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java]: Thanks for fixing this so quickly (I offered to provide a fix :) ).
Phoenix can actually be run completely without schemas - mostly for historical reasons.
Tables are allowed to have complex names (including . in that case).

**Comment by lhofhansl** [comment]
I suppose a way to test this is simply creating a table without a schema.
See PhoenixQueryRunner.copyTable(...)

**Comment by vincentpoon** [comment]
`TestPhoenixIntegrationSmokeTest#executeInPhoenix()` lets you execute sql directly against Phoenix (bypassing Presto).  You could create a table without schema with that, then call `SHOW SCHEMAS` through Presto.  Would that work?

**Comment by ebyhr** [comment]
@vincentpoon I already tried the approach, but it didn't return null schema.

**Comment by lhofhansl** [comment]
I propose we fix this and followup with a test. Right now this is broken, and many Phoenix users do not use schemas.
I'll try today now I can create a schema-less table from within Presto.

**Comment by ebyhr** [comment]
@lhofhansl While it returns null schema for `SHOW TABLES IN phoenix.default`, it doesn't return null for `SHOW SCHEMAS IN phoenix`.
